data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/v2/validator/PostAggregateAndProofsV2IntegrationTest.java

```diff
@@ -0,0 +1,156 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.beaconrestapi.v2.validator;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.when;
+import static tech.pegasys.teku.api.ValidatorDataProvider.PARTIAL_PUBLISH_FAILURE_MESSAGE;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_BAD_REQUEST;
+import static tech.pegasys.teku.infrastructure.http.HttpStatusCodes.SC_OK;
+import static tech.pegasys.teku.spec.SpecMilestone.ELECTRA;
+import static tech.pegasys.teku.spec.SpecMilestone.PHASE0;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import okhttp3.Response;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import tech.pegasys.teku.beaconrestapi.AbstractDataBackedRestAPIIntegrationTest;
+import tech.pegasys.teku.beaconrestapi.handlers.v2.validator.PostAggregateAndProofsV2;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.json.JsonTestUtil;
+import tech.pegasys.teku.infrastructure.json.JsonUtil;
+import tech.pegasys.teku.infrastructure.json.types.SerializableTypeDefinition;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.SpecMilestone;
+import tech.pegasys.teku.spec.TestSpecContext;
+import tech.pegasys.teku.spec.TestSpecInvocationContextProvider;
+import tech.pegasys.teku.spec.datastructures.operations.SignedAggregateAndProof;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+import tech.pegasys.teku.validator.api.SubmitDataError;
+
+@TestSpecContext(milestone = {PHASE0, ELECTRA})
+public class PostAggregateAndProofsV2IntegrationTest
+    extends AbstractDataBackedRestAPIIntegrationTest {
+
+  private DataStructureUtil dataStructureUtil;
+  private SpecMilestone specMilestone;
+  private SerializableTypeDefinition<List<SignedAggregateAndProof>> aggregateAndProofsListTypeDef;
+
+  @BeforeEach
+  void setup(final TestSpecInvocationContextProvider.SpecContext specContext) {
+    spec = specContext.getSpec();
+    specMilestone = specContext.getSpecMilestone();
+    startRestAPIAtGenesis(specMilestone);
+    dataStructureUtil = specContext.getDataStructureUtil();
+    aggregateAndProofsListTypeDef =
+        SerializableTypeDefinition.listOf(
+            spec.getGenesisSchemaDefinitions()
+                .getSignedAggregateAndProofSchema()
+                .getJsonTypeDefinition());
+  }
+
+  @TestTemplate
+  void shouldPostAggregateAndProofs_NoErrors() throws Exception {
```

Comment:
generally i think we try to do no capital directly after underscore? we should probably have a rule for it really...

Reply:
Removed the _ anyway